lib/charms/grafana_k8s/v0/grafana_dashboard.py

```diff
@@ -608,70 +608,32 @@ def _replace_template_fields(  # noqa: C901
-        # ${} templating and comparing the name to the list we built, replace it,
-        # otherwise, leave it alone.
-        #
-        # COS only knows about Prometheus and Loki.
-        for panel in panels:
-            if "datasource" not in panel or not panel.get("datasource"):
-                continue
-            if not existing_templates:
-                datasource = panel.get("datasource")
-                if type(datasource) == str:
-                    if "loki" in datasource:
-                        panel["datasource"] = "${lokids}"
-                    else:
-                        panel["datasource"] = "${prometheusds}"
-                elif type(datasource) == dict:
-                    # In dashboards exported by Grafana 9, datasource type is dict
-                    dstype = datasource.get("type", "")
-                    if dstype == "loki":
-                        panel["datasource"]["uid"] = "${lokids}"
-                    elif dstype == "prometheus":
-                        panel["datasource"]["uid"] = "${prometheusds}"
-                    else:
-                        logger.debug("Unrecognized datasource type '%s'; skipping", dstype)
-                        continue
-                else:
-                    logger.error("Unknown datasource format: skipping")
-                    continue
-            else:
-                if type(panel["datasource"]) == str:
-                    if panel["datasource"].lower() in replacements.values():
-                        # Already a known template variable
-                        continue
-                    # Strip out variable characters and maybe braces
-                    ds = re.sub(r"(\$|\{|\})", "", panel["datasource"])
-                    replacement = replacements.get(datasources[ds], "")
-                    if replacement:
-                        used_replacements.append(ds)
-                    panel["datasource"] = replacement or panel["datasource"]
-                elif type(panel["datasource"]) == dict:
-                    dstype = panel["datasource"].get("type", "")
-                    if panel["datasource"].get("uid", "").lower() in replacements.values():
-                        # Already a known template variable
-                        continue
-                    # Strip out variable characters and maybe braces
-                    ds = re.sub(r"(\$|\{|\})", "", panel["datasource"].get("uid", ""))
-                    replacement = replacements.get(datasources[ds], "")
-                    if replacement:
-                        used_replacements.append(ds)
-                        panel["datasource"]["uid"] = replacement
-                else:
-                    logger.error("Unknown datasource format: skipping")
-                    continue
+        # Find panels nested under rows
+        if "rows" in dict_content.keys():
+            rows = dict_content["rows"]
```

Comment:
Mimicking the pattern above,
```suggestion
        rows = dict_content.get("rows", {})
        if rows:
```